c++/src/kj/compat/tls.c++

```diff
@@ -248,16 +243,26 @@ public:
   }
 
   kj::Promise<size_t> tryRead(void* buffer, size_t minBytes, size_t maxBytes) override {
-    return tryReadInternal(buffer, minBytes, maxBytes, 0);
+    size_t alreadyDone = 0;
+    while (true) {
+      auto n = co_await sslCall([this,buffer,maxBytes]() { return SSL_read(ssl, buffer, maxBytes); });
+      if (n >= minBytes || n == 0) {
+        co_return alreadyDone + n;
+      } else {
+        buffer = reinterpret_cast<byte*>(buffer) + n;
+        minBytes -= n;
+        maxBytes -= n;
+        alreadyDone += n;
```

Comment:
Hope we have good test coverage for this

Reply:
It may be safer to avoid optimising in the same PR and instead do what the previous code did (recursion).

Reply:
> It may be safer to avoid optimising in the same PR and instead do what the previous code did (recursion).

we can't keep recursion with coroutines